packages/admin/admin/src/form/file/FileSelectListItem.tsx

```diff
@@ -86,21 +95,42 @@ export const FileSelectListItem = (inProps: FileSelectListItemProps) => {
         delete: deleteIcon = <Delete />,
         loading: loadingIcon = <ThreeDotSaving />,
         error: errorIcon = <Error fontSize="inherit" />,
+        filePreviewGenericFile: filePreviewGenericFileIcon = <FileIcon fontSize="inherit" color="inherit" />,
+        filePreviewLoading: filePreviewLoadingIcon = <ThreeDotSaving fontSize="inherit" color="inherit" />,
+        filePreviewError: filePreviewErrorIcon = <FileNotMenu fontSize="inherit" color="inherit" />,
     } = iconMapping;
 
-    if ("loading" in file && !file.name) {
-        return <Skeleton variant="rounded" animation="wave" {...slotProps?.skeleton} />;
-    }
+    const getFilePreviewContent = React.useCallback(() => {
```

Comment:
This callback has no arguments, so it can be a variable instead. Also, do you need memoization for this?